### ISSUE TITLE
Support multiple columns for TableCombo text

### ISSUE THREAD

**Issue by laeubi** [EclipseNebula/nebula#308]
Currently one needs to select one column that is responsible for display the text.
It would be good to have an alternative so the user has more control, I can think about the following things:

1. Support setting a Function<Object, String>, ToIntFunction<String> to perform conversion Object>String and int>String
2. Allow to set the column to -1 that results in simply concatenate all column text with a pipe ( C1 | C2 |  C3) as a result.

**Comment by wimjongman** [comment]
Christoph, we are happy with all your patches.

**Comment by lcaron** [comment]
Hi Christoph,

I had a look at this request this morning (thanks to quarantine & lockdown).

It is not as easy as it seems : the selected column is responsible for many thing: which text to display, which image to displayn, which presentation (font/foreground color/background color) to display...

So we need 5 functions (one for each purpose), or a class with 5 methods (getText(), getImage(), ...) and JFace has been designed for this.

What do you think ?

**Comment by laeubi** [comment]
I can imagine that's not that easy if one assumes ALL possibilities and combinations.
What I can think of is that one can supply a labelprovider for the Combo itself so one has tu return one image, one font, one text... that might even looks different from how the column wise rendering looks like if the combo table is open.

**Comment by laeubi** [comment]
One example might be I want to show a Table of currencies and tax rates that looks like this:

Country | Currency | Tax rate | Total|
DE | € | 19% | 2345,87 €
EN | $ | ...
....

When I selected a column I probably want to show the text field/selection in the following way:

(Flag image of the country) 2345,87 € (19%)

**Comment by lcaron** [comment]
Interesting, do you have in mind something like 
```
public class TableComboRenderer() {
    public String getText(int columnIndex) {...}
    public String getText(int columnIndex) {...}
   ...
}

```
?

**Comment by laeubi** [comment]
In code I would think about instead:

`tableColumnViewer.getTableCombo().setDisplayColumnIndex(1);`

id like something like:

`tableColumnViewer.setDisplayColumnRenderer(ILabelProvider);`

the label provider would then just be passed the selected row/item and returns a text/icon

**Comment by lcaron** [comment]
Ok i'll work on it, and tell you informed when the PR is ready

**Comment by lcaron** [comment]
Christoph,

Please tell me if it is ok for you : https://github.com/eclipse/nebula/pull/322 ?

Happy Easter

**Comment by lcaron** [comment]
Hi Christoph,

Can I have a feedback about the PR 322 please ?

Thanks,

Laurent

**Comment by lcaron** [comment]
Bug fixed in Nebula 2.5.0
